### PR TITLE
Article template

### PR DESCRIPTION
**Pull request by noslouch**
[ticket](https://jira.wnyc.org/browse/CMS-545)

## PR dependencies:
- remove black borders & other tweaks: nypublicradio/pattern-lab#68
- markup updates: nypublicradio/nypr-design-system#29

## notes
- [as per the spec](https://github.com/nypublicradio/cms/blob/master/docs/specs/2019-07-17-headless-routing.md#look-up-articles-using-html_path), articles are looked up using the `html_path` endpoint
- article sections are embedded in the URL, but also computed as metadata off of the `ancestry` key. when deriving an article's URL, we base it on the `html_url` value, stripping out the `section` segment, so that `https://cms.demo.nypr.digital/news/articles/lead-audio-test/` -> `articles/lead-audio-test/`. note the trailing slash :(
- since sections are part of the URL, we use them to construct an article URL when using the `<LinkTo />` component, so article urls take two arguments: `article.section.slug` and `article.path`

## blocks
StreamField blocks are supported using the `<ArticleBody />` component. An article model's `body` attribute (which should be an array of streamfield blocks) is passed in as the `block` argument, and the component will render the corresponding nested components (e.g. `<ArticleBody::paragraph />`) accordingly.

I've included a few basic components for the current supported article body component types: `paragraph`, `embed`, `image`, etc. To support a new type, simply add a new component `$ ember g component article-body/new-type --pod`, and `<ArticleBody />` will render the correct component when it receives that type from wagtail. If the component receives an unrecognized type, it'll skip.

### CODE REVIEW

File: app/components/article-body/code/component.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+
+export default Component.extend({
+  tagName: '',
+
+  nodes: computed('value', function() {
+    if (typeof FastBoot === 'undefined') {
+      const range = document.createRange();
+      return range.createContextualFragment(this.value.code);
```
Comment (mike-hearn) [inline app/components/article-body/code/component.js]: had no idea this was a thing, but it seems cool

**Comment by noslouch** [comment]
yeah I could see the `<ArticleBody/>` component getting refactored to a `<StreamField/>` component that has a more general purpose. but let's get there when we get there

File: app/components/article-body/code/component.js (hunk 1 above)
Comment (noslouch) [reply app/components/article-body/code/component.js]: oh yeah. contextual fragments are great. take a look at `app/utils/dom-fixer` if you really wanna see contextual fragments get busy

Comment (noslouch) [reply app/components/article-body/code/component.js]: oh yeah, contextual fragments are great. take a look at `/app/utils/dom-fixer` if you want to see more DOM stuff

File: app/routes/article.js [hunk 2]
```diff
@@ -28,49 +43,94 @@ export default Route.extend({
...
       publishedTime: model.publishedMoment.format(),
-      modifiedTime: model.modifiedMoment.format(),
+      modifiedTime: model.modifiedMoment.isValid() && model.modifiedMoment.format(),
       section: model.section.title,
       tags: model.displayTags,
       authors: model.authors,
       image: {
-        full: model.thumbnail640,
-        width: 640,
+        full: model.imageImage ? wagtailImageUrl([{id: model.leadImage.image}, 640, null, 'width'], {}) : null,
```
Comment (mike-hearn) [inline app/routes/article.js]: is this supposed to check against `model.leadImage` rather than `model.imageImage`

**Pull request merged**